hr_timesheet_sheet/models/account_analytic_line.py

```diff
@@ -8,62 +8,37 @@
 class AccountAnalyticLine(models.Model):
     _inherit = 'account.analytic.line'
 
-    sheet_id_computed = fields.Many2one(
-        comodel_name='hr_timesheet.sheet',
-        string='Sheet',
-        compute='_compute_sheet',
-        index=True,
-        ondelete='cascade',
-        search='_search_sheet'
-    )
     sheet_id = fields.Many2one(
         comodel_name='hr_timesheet.sheet',
         string='Sheet',
-        compute='_compute_sheet',
-        store=True,
     )
 
-    @api.depends('date', 'user_id', 'project_id', 'task_id', 'company_id',
-                 'sheet_id.date_start', 'sheet_id.date_end',
-                 'sheet_id.employee_id', 'sheet_id.company_id')
     def _compute_sheet(self):
         """Links the timesheet line to the corresponding sheet"""
         for timesheet in self:
-            if timesheet.sheet_id or not timesheet.project_id:
+            if not timesheet.project_id:
                 continue
-            sheets = self.env['hr_timesheet.sheet'].search(
-                [('date_end', '>=', timesheet.date),
-                 ('date_start', '<=', timesheet.date),
-                 ('employee_id.user_id.id', '=', timesheet.user_id.id),
-                 ('company_id', 'in', [timesheet.company_id.id, False]),
-                 ('state', '=', 'draft'),
-                 ])
-            if sheets:
-                timesheet.sheet_id_computed = sheets[0]
-                timesheet.sheet_id = sheets[0]
+            sheet = self.env['hr_timesheet.sheet'].search([
+                ('date_end', '>=', timesheet.date),
+                ('date_start', '<=', timesheet.date),
+                ('employee_id', '=', timesheet.employee_id.id),
+                ('company_id', 'in', [timesheet.company_id.id, False]),
```

Comment:
@astirpe IMO here simplify ` ('company_id', 'in', timesheet.company_id.ids),`

If not `timesheet.company_id.ids` then return [ ]